components/mediation-admin/org.wso2.carbon.message.processor/src/main/java/org/wso2/carbon/message/processor/service/MessageProcessorAdminService.java

```diff
@@ -755,4 +761,111 @@ public boolean validateAxis2ClientRepo(String input) throws AxisFault {
         }
     }
 
+    /**
+     * Gets the message from the associated queue.
+     *
+     * @param processorName message processor name.
+     * @return <code>message</code> returns message received from the queue as a string.
+     */
+    public String getMessage(String processorName) throws AxisFault {
+        SynapseConfiguration configuration = getSynapseConfiguration();
+        MessageConsumer messageConsumer = getMessageConsumer(configuration,processorName);
+        String message = null;
+
+        try {
+            message = getMessageAsString(messageConsumer);
+        } catch (AxisFault e) {
+            handleException(log, "Failed to get message :", e);
+        }
+
+        messageConsumer.cleanup();
+        return message;
+    }
+
+    private String getMessageAsString(MessageConsumer messageConsumer) throws AxisFault {
+        MessageContext messageContext;
+        String message = null;
+
+        if (messageConsumer.isAlive()) {
+            try {
+                messageContext = messageConsumer.receive();
+                message = messageContext.getEnvelope().toString();
+            } catch (SynapseException e) {
+                handleException(log, "Failed to get message :", e);
+            }
+        }
+        return  message;
+    }
+
+    /**
+     * Pops the message from the associated queue.
+     *
+     * @param processorName message processor name.
+     */
+    public void popMessage(String processorName) throws AxisFault {
+        SynapseConfiguration configuration = getSynapseConfiguration();
+        MessageConsumer messageConsumer = getMessageConsumer(configuration,processorName);
+
+        try {
+            popMessageFromQueue(messageConsumer);
+        } catch (SynapseException e) {
+            handleException(log, "Failed to get message : ", e);
```

Comment:
The error log is the same in all methods. Will be hard to debug issues. Shall we rename them accrodingly?

Reply:
I have changed the error logs to be more descriptive